Trie/TrieProj/Trie.cs

```diff
@@ -0,0 +1,154 @@
+﻿namespace TrieProj;
+
+/// <summary>
+///  Trie, also called prefix tree, data structure used for store associative array,
+///  with keys that usually are strings, by individual chars.
+/// </summary>
+public class Trie
+{
+    private TrieNode Root { get; }
+
+    /// <summary>
+    /// Constructor of <see cref="Trie"/> class instance 
+    /// </summary>
+    public Trie()
+    {
+        Root = new TrieNode();
+    }
+
+    /// <summary>
+    /// Adds a word in instance of <see cref="Trie"/> class by character
+    /// </summary>
+    public void Add(string word)
+    {
+        var current = Root;
+
+        foreach (var ch in word.ToCharArray())
+        {
+            if (current.Children.ContainsKey(ch))
+                current = current.Children[ch];
+            else
+            {
+                current.Children.TryAdd(ch, new TrieNode());
+                current = current.Children[ch]; 
+            }
+        }
+
+        current.IsWord = true;
+    }
+
+    /// <summary>
+    /// Checks if instance of <see cref="Trie"/> class contains word
+    /// </summary>
+    public bool Contains(string word)
+    {
+        var current = Root;
+
+        foreach (var ch in word.ToCharArray())
+        {
+            if (!current.Children.ContainsKey(ch))
+                return false;
+            var node = current.Children[ch];
+            current = node;
+        }
+
+        return current.IsWord;
+    }
+
+    /// <summary>
+    /// Auxiliary for <see cref="Remove(TrieNode, string, int)"/> method
+    /// </summary>
+    public void Remove(string word)
+    {
+        Remove(Root, word, 0);
+    }
+
+    /// <summary>
+    /// Recursive method for removing word from <see cref="Trie"/> class instance
+    /// </summary>
+    private static bool Remove(TrieNode current, string word, int index)
+    {
+        if (index == word.Length) {
+            if (!current.IsWord) {
```

Comment:
Внезапно "египетский" стиль скобок. В C# так не принято, надо так:
```suggestion
        if (index == word.Length) 
        {
            if (!current.IsWord) 
            {
```